src/ray/object_manager/ownership_based_object_directory.cc

```diff
@@ -80,11 +80,19 @@ ray::Status OwnershipBasedObjectDirectory::ReportObjectAdded(
   request.set_node_id(node_id.Binary());
 
   rpc_client->AddObjectLocationOwner(
-      request, [worker_id, object_id](Status status,
-                                      const rpc::AddObjectLocationOwnerReply &reply) {
+      request, [worker_id, object_id, node_id](
+                   Status status, const rpc::AddObjectLocationOwnerReply &reply) {
         if (!status.ok()) {
-          RAY_LOG(ERROR) << "Worker " << worker_id << " failed to add the location for "
-                         << object_id;
+          if (status.IsObjectNotFound()) {
```

Comment:
Again, when does it happen?

Reply:
When the object has been freed and therefore the reference has been deleted on the owner. See [this comment](https://github.com/ray-project/ray/pull/13841/files#r568717158), and definitely lmk if I'm missing something there.